crates/octez/src/async/rollup.rs

```diff
@@ -0,0 +1,228 @@
+use crate::unused_port;
+
+use super::endpoint::Endpoint;
+use anyhow::Result;
+use http::Uri;
+use std::{
+    path::{Path, PathBuf},
+    str::FromStr,
+};
+use tezos_crypto_rs::hash::SmartRollupHash;
+use tokio::process::{Child, Command};
+
+const DEFAULT_BINARY_PATH: &str = "octez-smart-rollup-node";
+
+#[derive(Clone, PartialEq, Debug)]
+pub enum RollupDataDir {
```

Comment:
i thought about that too. My rational was for the rollup to follow the same api with other components where the `temp dir` is supported (e.g. the node and client), and it wouldn't make sense for the rollup to support `temp dir` without specifying the path to the pre images. i agree this is a bit awkward but it can also just be a convenient api on top of the rollup.

so we either 
1. support temp dir that requires preimages 
2. do not support temp dir at all. 


cc @zcabter